src/domain.rs

```diff
@@ -490,7 +505,7 @@ impl<F> Domain<F> {
         0
```

Comment:
While reading through I did also just realize that this is probably unfortunate — we'd likely _want_ loom to also check our due time checking, probably by using some kind of mocked time. But again, a change for another time.